File: java/parser/src/main/java/gov/lbl/parser/parser/Equation_sectionVisitor.java
package gov.lbl.parser.parser;

import java.util.List;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Equation;
import gov.lbl.parser.domain.Equation_section;

import static java.util.stream.Collectors.toList;

public class Equation_sectionVisitor extends modelicaBaseVisitor<Equation_section> {
    @Override
    public Equation_section visitEquation_section(modelicaParser.Equation_sectionContext ctx) {
        Boolean initial = ctx.INITIAL() == null ? false : true;

        EquationVisitor equationVisitor = new EquationVisitor();
        List<Equation> equations = ctx.equation() == null ? null : ctx.equation()
                                                                    .stream()
                                                                    .map(equation -> equation.accept(equationVisitor))
                                                                    .collect(toList());
        return new Equation_section(initial, equations);
    }
}

File: jsParser/parser/Equation_sectionVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Equation_section = require('../domain/Equation_section')

const EquationVisitor = require('./EquationVisitor')

class Equation_sectionVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitEquation_section (ctx) {
    const initial = !!ctx.INITIAL()
    const equations = []

    if (ctx.equation()) {
      const equationVisitor = new EquationVisitor.EquationVisitor()
      ctx.equation().forEach(eqn => {
        equations.push(equationVisitor.visitEquation(eqn))
      })
    }

    return new Equation_section.Equation_section(initial, equations)
  }
};

Equation_sectionVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitEquation_section = this.visitEquation_section
exports.Equation_sectionVisitor = Equation_sectionVisitor
